### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by anatolii-zarytskyi**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/impl/AddOperationTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Test (expected = RuntimeException.class)
+    public void apply_addIncorrectQuantity_NotOk() {
+        operation.apply(fruitRecordDtoWithWrongQuantity);
+    }
+
+    @After
+    public void cleanMapDB() {
+        Storage.fruits.clear();
+    }
```
Comment (MakeDreamCode) [inline src/test/java/core/basesyntax/service/impl/AddOperationTest.java:72]: Why not @Befor?

**Review by MakeDreamCode — CHANGES_REQUESTED** [review]

**Review by MakeDreamCode — APPROVED** [review]

File: src/test/java/core/basesyntax/service/impl/WriteServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            + "apple,90" + System.lineSeparator();
+
+    @BeforeClass
+    public static void beforeClass() {
+        writerService = new WriteServiceImpl();
+    }
+
+    @Test
+    public void writeToFileWithCorrectInput_Ok() {
+        Assert.assertNotNull(CORRECT_PATH, FRUIT_DATA);
```
Comment (RomanDubovskyi) [inline src/test/java/core/basesyntax/service/impl/WriteServiceImplTest.java:23]: what does this method test?

File: src/test/java/core/basesyntax/service/impl/PurchaseOperationTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        operation.apply(fruitRecordDtoWithWrongQuantity);
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void apply_addIncorrectFruitName_NotOk() {
+        operation.apply(fruitRecordDtoWithWrongFruitName);
+    }
+
+    @After
+    public void cleanMapDB() {
```
Comment (RomanDubovskyi) [inline src/test/java/core/basesyntax/service/impl/PurchaseOperationTest.java:61]: you don't need to clear data twice per test method

File: src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    @Test
+    public void getReport_Ok() {
+        fruitService.addToStorage(recordDtos);
+        String actual = fruitService.getReport();
+        assertEquals(expected, actual);
+    }
+
+    @After
+    public void cleanMapDbBeforeStart() {
+        Storage.fruits.entrySet().clear();;
```
Comment (RomanDubovskyi) [inline src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java:71]:
```suggestion
        Storage.fruits.clear();
```

File: src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    private static final String expected = "fruit,quantity" + System.lineSeparator()
+            + "banana,10" + System.lineSeparator();
+    private final List<FruitRecordDto> recordDtos = List.of(
+            new FruitRecordDto(Operation.BALANCE, "banana", 10),
+                new FruitRecordDto(Operation.SUPPLY, "banana", 10),
+                new FruitRecordDto(Operation.RETURN, "banana", 10),
+                new FruitRecordDto(Operation.PURCHASE, "banana", 20));
+
+    @Before
+    public void clearDbBeforeStart() {
```
Comment (RomanDubovskyi) [inline src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java:30]: same here with before and after

File: src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FruitServiceImplTest {
+    private static FruitService fruitService;
+    private static final String expected = "fruit,quantity" + System.lineSeparator()
```
Comment (RomanDubovskyi) [inline src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java:21]: move this to the method, where you need it. Now it looks a bit confusing that you have several tests and initialize expected here

File: src/test/java/core/basesyntax/service/impl/FruitRecordDtoParserImplTest.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    public void parse_correctDataFromFile_Ok() {
+        List<String> dataFromFile = new ArrayList<>(Arrays.asList("type,fruit,quantity",
+                "b,banana,10",
+                "s,banana,20",
+                "p,banana,15"));
+        List<FruitRecordDto> actual = fruitRecordDtoParser.parse(dataFromFile);
+        List<FruitRecordDto> expected = List.of(new FruitRecordDto(Operation.BALANCE, "banana", 10),
+                new FruitRecordDto(Operation.SUPPLY, "banana", 20),
+                new FruitRecordDto(Operation.PURCHASE, "banana", 15));
+        assertEquals(expected.toString(), actual.toString());
```
Comment (RomanDubovskyi) [inline src/test/java/core/basesyntax/service/impl/FruitRecordDtoParserImplTest.java:32]:
```suggestion
        assertEquals(expected, actual));
```
I don't think we need to string for comparison

File: src/test/java/core/basesyntax/model/OperationTest.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    @Test
+    public void getOperationFromString_supply_Ok() {
+        expected = Operation.SUPPLY;
+        actual = Operation.getOperationFromString("s");
+        assertEquals(expected, actual);
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void getOperationFromString_incorrectOperation_NotOk() {
+        Operation.getOperationFromString("o");
```
Comment (RomanDubovskyi) [inline src/test/java/core/basesyntax/model/OperationTest.java:41]: declare o as variable invalid operation

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**